### PR TITLE
Analyzer Tests

### PR DESCRIPTION
**Pull request by marcauberer**
- Add tests for the analyzer component
- Text fixes in `IncompatibleTypesException` error message
- Upgrade Alpine Docker base image
- Upgrade Goreleaser action version
- Code optimizations

### CODE REVIEW

File: compiler/cpp/test/analyzer/AnalyzerTest.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    Analyzer analyzer = Analyzer(false, fileInput, jsonParser, "//", "/*", "*/");
+    // Execute test case
+    try {
+        analyzer.executeAnalysis();
+        if (param.errorMessage.length() > 0)
+            FAIL() << "Expected error message '" + param.errorMessage + "', but got no error";
+        SUCCEED();
+    } catch (IncompatibleTypesException &error) {
+        if (param.errorMessage.length() == 0)
+            FAIL() << "Expected no error, but got '" + std::string(error.what()) + "'";
```
Comment (Splines) [inline compiler/cpp/test/analyzer/AnalyzerTest.cpp:107]: "Expected no error, but got...", however the if statement is checking if there is no error message `param.errorMessage.length() == 0`.

Vice versa holds true for line 103.

---

Ah ok, got it now, the `param.errorMessage` is the error message that we expect...

File: compiler/cpp/test/test-files/analyzer/conditional-section-error-boolean-int/input-file.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Marc Auberer 2021. All rights reserved.
```
Comment (Splines) [inline compiler/cpp/test/test-files/analyzer/conditional-section-error-boolean-int/input-file.txt:1]: Other test files have a different copyright header.

**Review by Splines — APPROVED** [review]
Nice, love that you've testes all permutations.

File: compiler/cpp/test/analyzer/AnalyzerTest.cpp (hunk 1 above)
Comment (marcauberer) [reply compiler/cpp/test/analyzer/AnalyzerTest.cpp:107]: Exactly

File: compiler/cpp/test/test-files/analyzer/conditional-section-error-boolean-int/input-file.txt (hunk 2 above)
Comment (marcauberer) [reply compiler/cpp/test/test-files/analyzer/conditional-section-error-boolean-int/input-file.txt:1]: Doesn't matter, as it is only a test file

**Pull request merged**